deepCR/dataset.py

```diff
@@ -1,6 +1,114 @@
+        :param seed: fix numpy random seed to seed, for reproducibility.
+        """
+
+        np.random.seed(seed)
+        self.len_image = len(image)
+        self.len_mask = len(cr)
+        self.n_mask = n_mask
+        self.noise = noise
+        self.saturation = saturation
+
+        f_train = 1 - f_val
+        if part == 'train':
+            slice = np.s_[:int(self.len_image * f_train)]
+            slice_cr = np.s_[:int(self.len_mask * f_train)]
+        elif part == 'val':
+            slice = np.s_[int(self.len_image * f_train):]
+            slice_cr = np.s_[:int(self.len_mask * f_train)]
+        else:
+            slice = np.s_[0:]
+            slice_cr = np.s_[0:]
+
+        if sky is None:
+            sky = np.zeros(self.len_image)
+        elif type(sky) != np.ndarray:
+            sky = np.array([sky]*self.len_image)
+
+        self.image = image[slice]
+        self.cr = cr[slice_cr]
+        self.sky = sky[slice]
+        self.aug_sky = aug_sky
+        self.aug_img = (np.log10(aug_img[0]), np.log10(aug_img[1]))
+        self.norm = norm
+        self.percentile_limit = percentile
+
+        self.len_image = len(self.image)
+        self.len_mask = len(self.cr)
+
+    def get_cr(self):
+        """
+        Generate cosmic ray images from stacked dark frames.
+        Sample self.n_mask number of cr masks and add them together.
+        :return: (cr_img, cr_mask): sampled and added dark image containing only cr and accumulative cr mask
+        """
+        cr_id = np.random.randint(0, self.len_mask, self.n_mask)
+        crs = []; masks = []
+        for i in cr_id:
+            arr = np.load(self.cr[i])
+            crs.append(arr[0][None,:])
+            masks.append(arr[1][None,:])
+        masks = np.concatenate(masks).sum(axis=0) > 0
+        crs = np.concatenate(crs).sum(axis=0)
+        return crs, masks
+
+    def get_image(self, i):
+        data = np.load(self.image[i])
+        if len(data.shape) == 3:
+            return data[0], data[1]
+        else:
+            return data, np.zeros_like(data)
+
+    def __len__(self):
+        return self.len_image
+
+    def __getitem__(self, i):
+        cr, mask = self.get_cr()
+        image, ignore = self.get_image(i)
+        f_bkg_aug = self.aug_sky[0] + np.random.rand() * (self.aug_sky[1] - self.aug_sky[0])
+        f_img_aug = self.aug_img[0] + np.random.rand() * (self.aug_img[1] - self.aug_img[0])
+        f_img_aug = 10**f_img_aug
+        bkg = f_bkg_aug * self.sky[i]
+        img = (image + bkg) * f_img_aug + cr
+        scale = img.copy()
+        scale[scale < 1] = 1
+        scale = scale**0.5
+        if self.noise:
+            noise = np.random.normal(0, scale, image.shape)
+        else:
+            noise = np.zeros_like(image)
+        img += noise
+        img[img>self.saturation] = self.saturation
+
+        if self.norm:
+            limit = np.percentile(img, self.percentile_limit)
```

Comment:
I think you want to scale by the std of the background so you have zero median/unit variance.
But, as implemented, I do not think that the normmed version is returning what you want it to: by default `limit` will be the 50th-percentile (median) so that `clip` will be everything less than the original median. You then take the std of this clipped image and subtract off the median of this clipped image. 

Typically, iteratively one clips by std, like:

```
orig_img = img.copy()
for sigma in [5,4,4,3]:
   clip = img[img < (img.median() + img.std()*sigma)]
   img = clip

orig_img -= img.median()
orig_img /= img.std()
```